run-build.sh

```diff
@@ -59,6 +73,11 @@ while [[ $# > 0 ]]; do
     shift
 done
 
+# $args array may have empty elements in it.
+# The easiest way to remove them is to cast to string and back to array.
+temp="${args[@]}"
+args=($temp)
```

Comment:
Have you tested this with quoted parameters?

For example something like `-e "Hello World"` Should resolve to two elements in the `args` array, and not 3.